Added search command to the cli

W0lfw00d opened the pull request:
Trying to have a stab at issue [#2](https://github.com/vutran/dext-cli/issues/2)

W0lfw00d committed:
Add search command to the cli

W0lfw00d committed:
api.search now returns an array

adnasa commented on cli.js:
```diff
@@ -14,7 +14,11 @@ args.command(['search', 's'], 'Seach for plugins or themes.', (name, sub) => {
       spinner.succeed();
-      console.log(results);
+      if(results.length > 0) {
+        results.forEach(result => console.log(' - ' + result));
```
There are cleaner approach to this.

First: use reduce or map to build a list of strings. At least in the end, you can `console.log` __once__
Second: use string interpolation/templating coming with ES6

just an example

```js
const resultMessage = results.length
  ? results.reduce(result => `- ${result}`, '').join('\n')
  : 'no packages found'

console.log(chalk.yellow(resultMessage))
```

adnasa commented on cli.js:
```diff
@@ -6,6 +6,27 @@ const chalk = require('chalk');
+        results.forEach(result => console.log(' - ' + result));
+      } else {
+        console.log(chalk.yellow('no packages found'));
+      }
```
There are cleaner approach to this.

First: use reduce or map to build a list of strings. At least in the end, you can `console.log` **once**
Second: use string interpolation/templating coming with ES6

just an example

``` js
const resultMessage = results.length
  ? results.reduce(result => `- ${result}`, '').join('\n')
  : 'no packages found'

console.log(chalk.yellow(resultMessage))
```

adnasa reviewed (changes requested):
I have a minor change request

vutran reviewed (changes requested):
I've just published `dext-core-utils@v0.4.0`. Can you update that dependency and re-adjust the search command to handle the new search results structure ([view test](https://github.com/vutran/dext-core-utils/blob/develop/__tests__/api/index.js#L158))?

W0lfw00d committed:
- Update the dext-core-utils dep to v0.4.0
- Refactored the search command to add descriptions and only run console.log once

adnasa reviewed (approved):
LGTM code wise

vutran reviewed (changes requested):
We're getting close!
- Update `dext-core-utils` to `v0.4.1` with the latest hot patch
- We should also handle the case when user enters no search term in the CLI.
- Let's split the `results.reduce` line to multiple lines so it's not too long

``` js
resultsMessage = results.reduce(
  prev, result) => `${prev}- ${result.name}: ${result.desc}\n`, ''
);
```

W0lfw00d committed:
- Update the dext-core-utils dep to v0.4.1
- Don't search if no search term is provide

W0lfw00d commented:
I updated to v0.4.1 and now it will display "Search must be called with an search term" if none is provide.

vutran commented:
Awesome! This looks great!

vutran commented:
@W0lfw00d Would you like to become a maintainer of the Dext project?

W0lfw00d commented:
What would being a maintainer entail? 
I can see if I can make some more improvements, but I'm not sure what kind of time commitment (or skill level) I can bring to the table.

vutran commented:
No major commitment really. Just looking for extra heads to help on PR reviews when possible across our repos whenever you're free. 👍

W0lfw00d commented:
Okay, sign me up. 😄
